docs/avro.rst

```diff
@@ -145,22 +98,94 @@ simply dropped.  Had the new schema dropped the original field ``favorite_number
 forward compatible with the original user schema since we wouldn't know how to fill in the value for ``favorite_number``
 for the new data because the original schema did not specify a default value for that field.
 
+If the compatibility is set to ``FORWARD_TRANSITIVE`` (not just ``FORWARD``), then it means that consumers using all previousely registered schemas (not just the latest schema) can read data written by producers using the new schema.
+
 
 .. _avro-full_compatibility:
 
 Full Compatibility
 ^^^^^^^^^^^^^^^^^^
 
-Full compatibility means schemas are backward **and** forward compatible.
+``FULL`` compatibility means schemas are backward **and** forward compatible.
+This is when the change to the schema is both forward compatible and backward compatible. In some data formats, such as JSON, there are no full-compatible changes. Every modification is either only forward or only backward compatible. But in other data formats, like Avro, you can define fields with default values. In that case adding or removing a field with a default value is a fully compatible change.
 
 To support both previous use cases on the same data, we can evolve the schemas in a fully compatible way: old data can
 be read with the new schema, and new data can also be read with the old schema.
 
+If the compatibility is set to ``FULL_TRANSITIVE`` (not just ``FULL``), then it means that the new schema is forward and backward compatible with all previously registered schemas (not just the latest schema).
 
-|sr-long|
-^^^^^^^^^
 
-As you can see, when using Avro, one of the most important things is to manage its schemas and reason about how those
-schemas should evolve. :ref:`Confluent Schema Registry <schemaregistry_intro>` is built for exactly that purpose.
-You can find out the details on how to use it to store Avro schemas and enforce certain compatibility rules during
-schema evolution by looking at the :ref:`schemaregistry_api`.
+.. _avro-none_compatibility:
+
+No Compatibility Checking
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+``NONE`` compatibility type means schema compatibility checks are disabled.
+
+Sometimes we make incompatible changes.
+For example, modifying a field type from ``Number`` to ``String``.
+In this case, you will either need to upgrade all producers and consumers to the new schema version at the same time, or more likely – create a brand-new topic and start migrating applications to use the new topic and new schema, avoiding the need to handle two incompatible versions in the same topic.
+
+
+Transitive
+^^^^^^^^^^
+
+.. include:: includes/transitive.rst
+
+Transitive compatibility checking is important once you have more than two versions for a given subject.
+Refer to an `example of schema changes <https://github.com/confluentinc/schema-registry/issues/209>`__ which are incrementally compatible, but not transitively so.
+
+
+Summary Compatibility Types
```

Comment:
I think we should put this on top of this page, as tl;dr.

Reply:
Done